File: docker/send-test-email.py
import smtplib
from email.mime.text import MIMEText

# Connect to Greenmail SMTP
server = smtplib.SMTP('localhost', 3025)

# Create a test email (important)
msg = MIMEText("This is an urgent security alert! Your account may have been compromised. Please take immediate action.")
msg['Subject'] = 'URGENT: Security Alert - Immediate Action Required'
msg['From'] = 'security@example.com'
msg['To'] = 'test@localhost'

# Send the email
server.send_message(msg)
print("Test email 1 (important) sent!")

# Send another email (semi-important)
msg2 = MIMEText("Meeting scheduled for tomorrow at 2pm to discuss Q1 results.")
msg2['Subject'] = 'Meeting Request: Q1 Review'
msg2['From'] = 'manager@example.com'
msg2['To'] = 'test@localhost'
server.send_message(msg2)
print("Test email 2 (semi-important) sent!")

server.quit()
print("Done!")

File: examples/bits-imap-smtp/send-test-email.ts
import * as nodemailer from 'nodemailer';

async function main() {
  // Connect to Greenmail SMTP (use 127.0.0.1 to force IPv4)
  const transporter = nodemailer.createTransport({
    host: '127.0.0.1',
    port: 3025,
    secure: false,
  });

  // Create and send test email 1 (important)
  await transporter.sendMail({
    from: 'security@example.com',
    to: 'test@localhost',
    subject: 'URGENT: Security Alert - Immediate Action Required',
    text: 'This is an urgent security alert! Your account may have been compromised. Please take immediate action.',
  });
  console.log('Test email 1 (important) sent!');

  // Send another email (semi-important)
  await transporter.sendMail({
    from: 'manager@example.com',
    to: 'test@localhost',
    subject: 'Meeting Request: Q1 Review',
    text: 'Meeting scheduled for tomorrow at 2pm to discuss Q1 results.',
  });
  console.log('Test email 2 (semi-important) sent!');

  transporter.close();
  console.log('Done!');
}

main().catch(console.error);
